### ISSUE TITLE
Plugin development:  Using vagrant

### ISSUE THREAD

**Issue by drmikecrowe** [xxh/xxh#133]
I found the `xde` development a bit overkill for simple plugin development.  Instead, I developed the process below using [Vagrant](https://www.vagrantup.com/) and `virtualbox`.  If this is of interest, I can submit a PR or wiki updates if you like.

**For community:**
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

## Functionality

You can start the VM using the following:

```sh
vagrant up
```

My folder looks like this with my two plugins I wanted to develop:

```
-rwxrwxr-x 1 mcrowe mcrowe  940 Feb 13 11:32 start.sh*
-rw-rw-r-- 1 mcrowe mcrowe 3020 Feb 13 08:05 Vagrantfile
drwxrwxr-x 4 mcrowe mcrowe 4096 Feb 13 10:08 xxh-plugin-bash-bashit/
drwxrwxr-x 4 mcrowe mcrowe 4096 Feb 13 08:56 xxh-plugin-prerun-dotbare/
```
The `start.sh` script starts an `xxh` session to the VM and maps in any `xxh-` folders in the session.  So, for my development, I simply ran:

```sh
./start.sh
```

and was able to test a clean install.  To reset everything, you simply do:

```sh
vagrant destroy && vagrant up
```

## Details

`Vagrantfile`:
```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "ubuntu/jammy64"
end
```

`start.sh`:
```sh
#!/usr/bin/env bash

SHELL=${1:-bash}

VAGRANT_IP="$(VBoxManage showvminfo $(cat .vagrant/machines/default/virtualbox/id) --details  | grep 'name = ssh' | cut -b 18- | egrep -o 'host ip = [0-9.]*' | cut -b 11-)"
VAGRANT_PORT="$(VBoxManage showvminfo $(cat .vagrant/machines/default/virtualbox/id) --details  | grep 'name = ssh' | cut -b 18- | egrep -o 'host port = [0-9]*' | cut -b 13-
)"
SSH="-i .vagrant/machines/default/virtualbox/private_key -o PasswordAuthentication=no vagrant@$VAGRANT_IP -p $VAGRANT_PORT"

TESTING=""
CURR_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"

for item in $(shopt -s dotglob && cd $CURR_DIR && find . -type d -name 'xxh-*'); do
    TODO="$(basename $item)"
    TESTING="+RI $TODO+path+$CURR_DIR/$TODO $TESTING"
done

echo "Starting with params: $TESTING"
ssh $SSH "ls -l"
if [ $? != 0 ]; then
    ssh-keygen -f "$HOME/.ssh/known_hosts" -R "[$VAGRANT_IP]:$VAGRANT_PORT"
fi

xxh $SSH +s $SHELL $TESTING
```

**Comment by anki-code** [comment]
Hi! It will be great to have the repository with this way of testing with step by step guide/example and we can add the link into [development section](https://github.com/xxh/xxh#development).

**Comment by anki-code** [comment]
Closed #134
